float32 centroid1_x
float32 centroid1_y
float32 centroid2_x
float32 centroid2_y
float32 angle
